link/uprobe.go

```diff
@@ -83,32 +86,10 @@ func OpenExecutable(path string) (*Executable, error) {
 		return nil, fmt.Errorf("path cannot be empty")
 	}
 
-	f, err := os.Open(path)
```

Comment:
We don't return an error when the given path doesn't exist, which seems suboptimal. We could stat the file, but then we'd get a TOCTOU because we later on need to `Open` and the binary might have changed. Does that matter? Seems like the tracefs api only deals with paths as well, which implies it has the same problem. Is there a way to avoid paths with tracefs?

One workaround would be to still `Open` here and put `*os.File` into Executable. But then we leak `*os.File`. Do we need `Executable.Close()`?

Reply:
I thought about this yesterday :D 

> We don't return an error when the given path doesn't exist, which seems suboptimal. We could stat the file, but then we'd get a TOCTOU because we later on need to Open and the binary might have changed. Does that matter? Seems like the tracefs api only deals with paths as well, which implies it has the same problem. Is there a way to avoid paths with tracefs?

I think pmu has the same problem (perf event is opened on the given executable path), so if the binary changes it will break anyway (or, if the check has never been done, it will fail regardless)

Reply:
Hm, okay. And there is no API that takes an FD?

I'd still prefer to stat here in addition to your current approach so that we return an error early.

Reply:
> Hm, okay. And there is no API that takes an FD?

https://man7.org/linux/man-pages/man2/perf_event_open.2.html this is the API we are using :D (ctrl+f uprobe_path)

> I'd still prefer to stat here in addition to your current approach so that we return an error early.

ok, I'll add that check early